### PR TITLE
Add 'extras' to groupParticipantsUpdate

### PR DESCRIPTION
**Pull request by JonnyShuali**
The feature added a new object inside groupParticipantsUpdate object result, which contains the invite code and expiration timestamp while adding a new participant to a group.

### CODE REVIEW

**Comment by ravindu01manoj** [comment]
fix typo

**Comment by ravindu01manoj** [comment]
```ts

return participantsAffected.map(p => { return {status: p.attrs.error || 200,jid: p.attrs.jid,extras: action == 'add' ? WABinary_1.getBinaryNodeChildren(p, 'add_request').map((d)=> d.attrs) : false}})

```

**Comment by ravindu01manoj** [comment]
```ts
return participantsAffected.map(p => {
				return { status: p.attrs.error || 200, jid: p.attrs.jid, extras: action == 'add' && p.attrs.error ? getBinaryNodeChildren(p, 'add_request').map((d)=> d.attrs) : false}
			})
```

**Comment by JonnyShuali** [comment]
> fix typo

where?

**Comment by ravindu01manoj** [comment]
> ```ts
> return participantsAffected.map(p => {
> 				return { status: p.attrs.error || 200, jid: p.attrs.jid, extras: action == 'add' && p.attrs.error ? getBinaryNodeChildren(p, 'add_request').map((d)=> d.attrs) : false}
> 			})
> ```

can you update your pull request with this code

**Comment by JonnyShuali** [comment]
should be ok now

File: src/Socket/groups.ts [hunk 1]
```diff
@@ -100,9 +100,9 @@ export const makeGroupsSocket = (config: SocketConfig) => {
 			const node = getBinaryNodeChild(result, action)
 			const participantsAffected = getBinaryNodeChildren(node!, 'participant')
 			return participantsAffected.map(p => {
-				return { status: p.attrs.error || 200, jid: p.attrs.jid }
+				return { status: p.attrs.error || 200, jid: p.attrs.jid, extras: action == 'add' && p.attrs.error ? getBinaryNodeChildren(p, 'add_request').map((d)=> d.attrs) : false}
```
Comment (adiwajshing) [inline src/Socket/groups.ts:103]: Instead of putting this ternary operation here -- may be cleaner to put it in a line before the return statement

Comment (JonnyShuali) [reply src/Socket/groups.ts:103]: By my opinion chaning it to 
`			participantsAffected.map(participant => {
				let data = {status: participant.attrs.error || 200, jid: participant.attrs.jid}
				if (action == 'add' && participant.attrs.error) {
					data["extras"] = getBinaryNodeChildren(participant, 'add_request').map((d)=> d.attrs)
				}
				return data;
			})`
makes it harder to read than ternary operation. What do you think?

**Comment by JonnyShuali** [comment]
any updates?

**Comment by mustafamasvi** [comment]
any updates?

**Pull request closed without merging**